crates/subspace-archiving/src/reconstructor.rs

```diff
@@ -0,0 +1,246 @@
+use crate::archiver::{Segment, SegmentItem};
+use crate::utils;
+use parity_scale_codec::Decode;
+use reed_solomon_erasure::galois_16::ReedSolomon;
+use std::mem;
+use subspace_core_primitives::{
+    ArchivedBlockProgress, LastArchivedBlock, Piece, RootBlock, PIECE_SIZE, SHA256_HASH_SIZE,
+};
+use thiserror::Error;
+
+/// Reconstructor-related instantiation error.
+#[derive(Debug, Error, Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash)]
+pub enum ReconstructorInstantiationError {
```

Comment:
Since we're already under the reconstructor context, I'd simply use `InstantiationError` and `Error`, just a personal pref.

Reply:
It's no biggie. Since we already have all these _generic_`Error`s, I'm not personally against adding another one :P.